### PR TITLE
(v2.5.x) Fix CI upstream tests

### LINKED ISSUE #1178: 'test-asciidoctor-upstream' CI job not working due to torquebox domain expiration
[torquebox domain expired](https://github.com/torquebox/jruby-maven-plugins/issues/124) and causes upstream test to fail when installing the Asciidoctor upstream version locally.

```
Warning:  Could not validate integrity of download from https://rubygems-proxy.torquebox.org/releases/de/saumya/mojo/gem-maven-plugin/1.1.8/gem-maven-plugin-1.1.8.jar
org.eclipse.aether.transfer.ChecksumFailureException: Checksum validation failed, expected <!doctype but is 25650cb6d33c67ede12ee2fffa6afb12bc88c196
    at org.eclipse.aether.connector.basic.ChecksumValidator.validateExternalChecksums (ChecksumValidator.java:174)
    at org.eclipse.aether.connector.basic.ChecksumValidator.validate (ChecksumValidator.java:103)
    at org.eclipse.aether.connector.basic.BasicRepositoryConnector$GetTaskRunner.runTask (BasicRepositoryConnector.java:460)
    at org.eclipse.aether.connector.basic.BasicRepositoryConnector$TaskRunner.run (BasicRepositoryConnector.java:364)
    at org.eclipse.aether.util.concurrency.RunnableErrorForwarder$1.run (RunnableErrorForwarder.java:75)
    at java.util.concurrent.ThreadPoolExecutor.runWorker (ThreadPoolExecutor.java:1136)
    at java.util.concurrent.ThreadPoolExecutor$Worker.run (ThreadPoolExecutor.java:635)
    at java.lang.Thread.run (Thread.java:833)
Warning:  Could not validate integrity of download from https://rubygems-proxy.torquebox.org/releases/org/codehaus/plexus/plexus-utils/1.1/plexus-utils-1.1.jar
org.eclipse.aether.transfer.ChecksumFailureException: Checksum validation failed, expected <!doctype but is 25650cb6d33c67ede12ee2fffa6afb12bc88c196
    at org.eclipse.aether.connector.basic.ChecksumValidator.validateExternalChecksums (ChecksumValidator.java:174)
    at org.eclipse.aether.connector.basic.ChecksumValidator.validate (ChecksumValidator.java:103)
    at org.eclipse.aether.connector.basic.BasicRepositoryConnector$GetTaskRunner.runTask (BasicRepositoryConnector.java:460)
    at org.eclipse.aether.connector.basic.BasicRepositoryConnector$TaskRunner.run (BasicRepositoryConnector.java:364)
    at org.eclipse.aether.util.concurrency.RunnableErrorForwarder$1.run (RunnableErrorForwarder.java:75)
    at java.util.concurrent.ThreadPoolExecutor.runWorker (ThreadPoolExecutor.java:1136)
    at java.util.concurrent.ThreadPoolExecutor$Worker.run (ThreadPoolExecutor.java:635)
    at java.lang.Thread.run (Thread.java:833)
```

### PR DESCRIPTION
**Pull request by abelsromero**
## Kind of change

- [x] Bug fix
- [ ] New non-breaking feature
- [ ] New breaking feature
- [ ] Documentation update
- [ ] Build improvement

## Description

What is the goal of this pull request?

TLDR; Fix CI upstream tests, same as https://github.com/asciidoctor/asciidoctorj/pull/1179 but for v2.5.x branch.

How does it achieve that?

Since we need compatibility with Java 8, this PR only does minimal changes (unlike https://github.com/asciidoctor/asciidoctorj/pull/1179). Simply remove the `gem` packaging.

Also bump jruby version, not strictly necessary, but I'd rather be on latests versions, you can't never be too paranoid about security.

Are there any alternative ways to implement this?

Same considerations as in https://github.com/asciidoctor/asciidoctorj/pull/1179.

Are there any implications of this pull request? Anything a user must know?

Yes, always keep your CI healthy green_circle

## Issue

If this PR fixes an open issue, please add a line of the form:

Fixes #1178 

## Release notes

Please add a corresponding entry to the file CHANGELOG.adoc

### CODE REVIEW

**Comment by robertpanzer** [comment]
I had to remove the pluginRepository to get it running locally:
```diff
diff --git a/ci/asciidoctor-gem-installer.pom b/ci/asciidoctor-gem-installer.pom
index e03ca15..67230bc 100644
--- a/ci/asciidoctor-gem-installer.pom
+++ b/ci/asciidoctor-gem-installer.pom
@@ -14,12 +14,6 @@
             <url>http://rubygems-proxy.torquebox.org/releases</url>
         </repository>
     </repositories>
-    <pluginRepositories>
-        <pluginRepository>
-            <id>rubygems-releases</id>
-            <url>https://rubygems-proxy.torquebox.org/releases</url>
-        </pluginRepository>
-    </pluginRepositories>
     <properties>
         <project.build.sourceEncoding>UTF-8</project.build.sourceEncoding>
     </properties>
```

**Comment by abelsromero** [comment]
Nevermind this branch, I am looking at the `main` branch in my fork. It's driving me crazy the nightly scheduled run is successful but the PRs are not :hankey:

**Comment by abelsromero** [comment]
Main branch is completed (if only for improvements), back to this one, I will be testing in my fork, don't worry if there's no activity here. I'll mark as ready when ready.

**Comment by abelsromero** [comment]
At the end ported the formatting improvements from main branch and it works fine now.

**Comment by robertpanzer** [comment]
Thanks a lot!
